plugin-server/src/worker/ingestion/event-pipeline/3-processPersonsStep.ts

```diff
@@ -0,0 +1,29 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { Person } from '../../../types'
+import { normalizeEvent } from '../../../utils/event'
+import { updatePersonState } from '../person-state'
+import { parseEventTimestamp } from '../timestamps'
+import { EventPipelineRunner, StepResult } from './runner'
+
+export async function processPersonsStep(
+    runner: EventPipelineRunner,
+    pluginEvent: PluginEvent,
+    person: Person | undefined
+): Promise<StepResult> {
+    const event = normalizeEvent(pluginEvent)
+    const timestamp = parseEventTimestamp(event, runner.hub.statsd)
+
+    const personInfo: Person | undefined = await updatePersonState(
+        event,
+        event.team_id,
+        String(event.distinct_id),
+        timestamp,
+        runner.hub.db,
+        runner.hub.statsd,
+        runner.hub.personManager,
+        person
+    )
+
+    return runner.nextStep('prepareEventStep', event, personInfo)
```

Comment:
Since we always need a person in the future steps and how it's better if person lookups happen in order (person properties will more naturally change on events), consider looking up the person from redis cache here & make prepareEventStep not need to deal with it being `undefined` (i.e. move https://github.com/PostHog/posthog/blob/f541b58242d827e68db5e507425caf5e225e5fed/plugin-server/src/worker/ingestion/process-event.ts#L221-L228 here).

Reply:
The redis cache is IMO sort of unneeded and should be removed completely - it didn't do much before given we were calling `fetchPerson` up to 6 times per event.

Now for most vanilla events we only fetch it once in the `emitToBuffer` step.

Reply:
sure we can remove the cache, my comment was more about the fact that lets make the person step return the person always and later steps not need to think about it.

Reply:
Not sure I understand. The `1-emitBufferStep` already always fetches person and it's used later in the pipeline.